### PR TITLE
Add Amazonbot and simplify some bot regexes

### PR DESCRIPTION
**Pull request by MichaIng**
### Description:

Amazonbot is Amazon's web crawler: https://developer.amazon.com/support/amazonbot
```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/600.2.5 (KHTML, like Gecko) Version/8.0.2 Safari/600.2.5 (Amazonbot/0.1; +https://developer.amazon.com/support/amazonbot)
```

The third option of the Baiduspider regex supersedes the first, and hence both are combined. While regexes are checked case-insensitive, it is changed to capital B for consistency with the other patterns, which have correct cases consequently.

All cases of leading and tailing `(...)?` optional strings have been removed, as they don't have any effect on whether a string matches or not. Optional characters are only relevant when being surrounded by non-optional characters.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: regexes/bots.yml [hunk 1]
```diff
@@ -164,7 +164,7 @@
     name: '2.0Promotion GbR'
     url: 'http://www.backlinktest.com'
 
-- regex: 'baiduspider(-image)?|baidu Transcoder|baidu.*spider'
+- regex: '[Bb]aidu.*spider|baidu Transcoder'
```
Comment (sanchezzzhak) [inline regexes/bots.yml:167]: `baidu.*spider|baidu Transcoder`

all regular expressions are executed with the flag `i`

https://github.com/matomo-org/device-detector/blob/975c63703f76e0831332b8310b40b0c9113dd003/Parser/AbstractParser.php#L248-L253

Comment (MichaIng) [reply regexes/bots.yml:167]: Ah okay, I wasn't expecting it else all other regexes have correct capitalisation. I'll keep the capital `B` then, as this is really correct for all known Baiduspider user agents.

I also recognised that `baidu.*spider` matched our two examples already due to the URL `http://www.baidu.com/search/spider.html`, even with lower case `b`. So it remains a little regex simplification.

**Review by sanchezzzhak — DISMISSED** [review]

**Comment by MichaIng** [comment]
Ah wait, sorry, now I found a missing user agent:
```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/600.2.5 (KHTML, like Gecko) Version/8.0.2 Safari/600.2.5 (Amazonbot/0.1; +https://developer.amazon.com/support/amazonbot)
```
Sorry for the back and forth, background is that two pages got a several thousand percent increase, clearly associated to bot activity, so I checked our access logs and compared user agents.

File: regexes/bots.yml [hunk 2]
```diff
@@ -61,6 +61,13 @@
     name: 'Alexa Internet'
     url: 'https://www.alexa.com'
 
+- regex: 'Amazonbot'
+  name: 'Amazon web crawler'
```
Comment (sanchezzzhak) [inline regexes/bots.yml:65]: * Set name `Amazon Bot`
* You need to add test to file device-detector/Tests/fixtures/bots.yml

You can get test fixture via the console
To do this, run the command
```shell
cd device-detector/misc
php test.php "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/600.2.5 (KHTML, like Gecko) Version/8.0.2 Safari/600.2.5 (Amazonbot/0.1; +https://developer.amazon.com/support/amazonbot)"
```

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

File: regexes/bots.yml (hunk 2 above)
Comment (MichaIng) [reply regexes/bots.yml:65]:
```
PHP Warning:  require(/mnt/dietpi_userdata/device-detector/misc/../vendor/autoload.php): Failed to open stream: No such file or directory in /mnt/dietpi_userdata/device-detector/misc/test.php on line 13
PHP Fatal error:  Uncaught Error: Failed opening required '/mnt/dietpi_userdata/device-detector/misc/../vendor/autoload.php' (include_path='.:/usr/share/php') in /mnt/dietpi_userdata/device-detector/misc/test.php:13
Stack trace:
#0 {main}
  thrown in /mnt/dietpi_userdata/device-detector/misc/test.php on line 13
```
I guess this needs to be done within the Matomo directory? Found `/matomo/vendor/matomo/device-detector`, but it does not contain the `misc/test.php`. Hehe sorry I'm a bit inexperienced with PHP development. I'll try to apply the test manually.

Comment (MichaIng) [reply regexes/bots.yml]: Done. Btw, is there a specific reason why in `regexes/bots.yml`, all strings are single quoted while in `Tests/fixtures/bots.yml`, those are not?

Comment (sanchezzzhak) [reply regexes/bots.yml]: > Btw, is there a specific reason why in regexes/bots.yml, all strings are single quoted while in Tests/fixtures/bots.yml, those are not?

There are many special characters for a'string in regular expressions, so they are in quotes,
also all regex are written by hand.
Most all fixtures are created via `misc/test.php` or sorted `misc/sortTestFixtureFiles.php` where extra quotes are removed.

**Review by sanchezzzhak — APPROVED** [review]

File: regexes/bots.yml (hunk 2 above)
Comment (sanchezzzhak) [reply regexes/bots.yml]: @MichaIng Thx you for PR.

Comment (MichaIng) [reply regexes/bots.yml]: Thanks for the quick review and merged 👍.

**Pull request merged**